### PR TITLE
Avoid copying object if TypeMeta is already correct.

### PR DESCRIPTION
**Pull request by markusthoemmes**
As per title. This short-circuits the copy code if the TypeMeta is already correct. Less allocations and stuff :slightly_smiling_face: 

/assign @mattmoor

### CODE REVIEW

File: controller/helper.go [hunk 1]
```diff
@@ -34,12 +37,23 @@ func EnsureTypeMeta(f Callback, gvk schema.GroupVersionKind) Callback {
...
+			return
+		}
+
+		// If TypeMeta is already what we want, exit early.
+		if accessor.GetAPIVersion() == apiVersion && accessor.GetKind() == kind {
+			f(typed)
+			return
+		}
+
 		// We need to populated TypeMeta, but cannot trample the
```
Comment (whaught) [inline controller/helper.go:52]:
```suggestion
		// We need to populate TypeMeta, but cannot trample the
```

File: controller/helper.go [hunk 2]
```diff
@@ -23,8 +23,11 @@ import (
 	"knative.dev/pkg/kmeta"
 )
 
+// Callback is a function that is passed to an informer's event handler.
 type Callback func(interface{})
 
+// EnsureTypeMeta augments the passed in callback, ensuring that all objects that pass
```
Comment (whaught) [inline controller/helper.go:29]:
```suggestion
// EnsureTypeMeta augments the passed-in callback, ensuring that all objects that pass
```

Comment (whaught) [reply controller/helper.go:29]: alternative "`callback parameter`"

**Comment by whaught** [comment]
> controller/helper.go

I'm surprised the [coverage run](https://storage.cloud.google.com/knative-prow/pr-logs/pull/knative_pkg/1561/pull-knative-pkg-go-coverage/1287667600452489216/artifacts/line-cov.html#file0) didn't catch the new logic?

**Comment by markusthoemmes** [comment]
Cause I didn't add a test for it :joy:. Done that now @whaught

**Pull request merged**